Change `tuist scaffold list` output to be readable by grep

mollyIV opened the pull request:
Resolves https://github.com/tuist/tuist/issues/1147

### Short description 📝

The purpose of this pull request is to make an output of `tuist scaffold list` command readable by grep. 
To make the review process easier (there's already quite a few changes), I'd like to add option for changing the output to JSON in a separate pull request.

### Solution 📦

As the solution, I introduced a new `TextTable` struct which allows for grepable table rendering. The API:  

```swift
let table = TextTable<Record>{[
  TextTable.Column(title: "Key", value: $0.key),
  TextTable.Column(title: "Value", value: $0.value)
]}

let data = [
  Record(key: "Key 1", value: 0),
  Record(key: "Key 2", value: 1),
]

let rendered = table.render(data)
```

> I took the inspiration for the API from [TextTable](https://github.com/cfilipov/TextTable) open source project.

The `TextTable` is used under the hood of `ListService`, to prepare the table about the templates before actual rendering to the console. 

I am very opened to your improvement suggestions related to both `TextTable` component and the integration part 😊 

### Implementation 👩‍💻👨‍💻

- [x] Introduced `TextTable`
- [x] Integrated `TextTable` into `ListService`
- [x] Unit Tests

### Testing 

```bash
$ scaffold list
Name     Description     
───────  ────────────────
default  Default template
swiftui  SwiftUI template
default  Default template
swiftui  SwiftUI template

```

pepibumur commented on Sources/TuistKit/Utils/TextTable.swift:
```diff
@@ -0,0 +1,107 @@
+import Foundation
+
+struct TextTable<T> {
```
Considering this is not Tuist-specific, what about moving it to `TuistSupport`? I really like the interface of this one btw :)

fortmarek replied:
+1 - I'd also add some simple documentation for it if we move it to `TuistSupport`

mollyIV replied:
Sure! I moved the `TextTable` to `TuistSupport` and added the documentation ✔️

mollyIV commented:
Hey @pepibumur , @fortmarek 👋 

Thanks a lot for a quick review and your comments. I moved `TextTable` to `TuistSupport` and added the documentation ✔️ Let me know if there's anything else or we can just merge the pull request 🚀 

Thanks!

pepibumur commented:
Thanks a lot @mollyIV 😀. I think we can merge once CI is green.
